gramps/gen/lib/grampstype.py

```diff
@@ -165,41 +179,27 @@ def __set_tuple(self, value):
         self.__value = val
         self.__string = strg
 
-    def __set_int(self, value):
+    @set.register
+    def set_int(self, value: int):
         "Set the value/string properties from an integer."
         self.__value = value
         self.__string = ""
 
-    def __set_instance(self, value):
-        "Set the value/string properties from another grampstype."
-        self.__value = value.value
-        if self.__value == self._CUSTOM:
-            self.__string = value.string
-        else:
-            self.__string = ""
-
-    def __set_str(self, value):
+    @set.register
+    def set_str(self, value: str):
```

Comment:
This should be private.

Reply:
This has been addressed in the latest commit